### PR TITLE
feat: adds document intelligence to bicep infrastructure

### LINKED ISSUE #448: Add bicep code for Azure Document Intelligence.
Currently there's no bicep code for deploying Azure Document Intelligence resource required for reading documents.

### PR DESCRIPTION
**Pull request by liammoat**
This pull request introduces Azure AI Document Intelligence to the infrastructure. The changes includes:

- Adding new parameters, modules, and outputs to support the Azure AI Document Intelligence resource in the `main.bicep` file.
- `storekeys.bicep` has been modified to store the Document Intelligence key in the Azure Key Vault. This is loaded in `environment.py` using the `from_keyvault` method.

Closes #448

### CODE REVIEW

File: infra/main.bicep [hunk 1]
```diff
@@ -165,6 +171,20 @@ module openai './shared/cognitiveservices.bicep' = {
   }
 }
 
+module documentIntelligence './shared/cognitiveservices.bicep' = {
+  name: azureAIDocumentIntelligenceResourceName
+  scope: rg
+  params: {
+    name: azureAIDocumentIntelligenceResourceName
+    location: location
```
Comment (ross-p-smith) [inline infra/main.bicep:179]: Things may have changed, is document intelligence available in all regions? Could we get in problems if someone picks a region that it doesn't exist? I believe we do some checking in this repo for Semantic Search as that is not available everywhere

Comment (liammoat) [reply infra/main.bicep]: Good spot, thanks @ross-p-smith. I have updated this to align with the approach taken for Semantic Search.

https://github.com/microsoft/rag-experiment-accelerator/pull/599/commits/b0d7fb8b655461f260dea30f2bfaf2a518bfd6b9

**Comment by liammoat** [comment]
@ritesh-modi can you see any reason login is failing on the pull_request pipeline? https://github.com/microsoft/rag-experiment-accelerator/actions/runs/10082305075/job/27876236056?pr=599

**Review by julia-meshcheryakova — APPROVED** [review]
LGTM

**Pull request merged**